bug: unable to import r2 bucket with v5 - 500 internal server error

**What happend:**  
The new v5 provider is unable to handle r2 bucket imports. TF/ToFu prepares the import but then crashes due to a 500 HTTP server resonse.

**Config:**  

```hcl
locals {
  account_id = "my-id"
}

terraform {
  required_version = ">= 1.0"
  required_providers {
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 5.0"
    }
  }
}

provider "cloudflare" {
  api_key = "my-token"
  email   = "my-mail"
}

resource "cloudflare_r2_bucket" "hegerdes-mastodon" {
  account_id = local.account_id
  name       = "hegerdes-mastodon"
}
```

The command `tofu import cloudflare_r2_bucket.hegerdes-mastodon my-id/hegerdes-mastodon` errors.

**TF Debug log:**

<details>
  <summary>Show Logs</summary>
  
  ```txt
2025-01-30T18:21:00.357+0100 [INFO]  OpenTofu version: 1.9.0
2025-01-30T18:21:00.358+0100 [DEBUG] using github.com/hashicorp/go-tfe v1.36.0
2025-01-30T18:21:00.358+0100 [DEBUG] using github.com/opentofu/hcl/v2 v2.0.0-20240814143621-8048794c5c52
2025-01-30T18:21:00.358+0100 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2025-01-30T18:21:00.358+0100 [DEBUG] using github.com/zclconf/go-cty v1.14.4
2025-01-30T18:21:00.358+0100 [INFO]  Go runtime version: go1.22.8
2025-01-30T18:21:00.358+0100 [INFO]  CLI args: []string{"C:\\Users\\Henrik\\Tools\\tofu.exe", "import", "cloudflare_r2_bucket.hegerdes-mastodon", "my-id/hegerdes-mastodon", "-no-color"}
2025-01-30T18:21:00.364+0100 [DEBUG] Attempting to open CLI config file: C:\Users\Henrik\AppData\Roaming\terraform.rc
2025-01-30T18:21:00.364+0100 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2025-01-30T18:21:00.365+0100 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2025-01-30T18:21:00.365+0100 [DEBUG] ignoring non-existing provider search directory C:\Users\Henrik\AppData\Roaming\terraform.d\plugins
2025-01-30T18:21:00.366+0100 [DEBUG] ignoring non-existing provider search directory C:\Users\Henrik\AppData\Roaming\HashiCorp\Terraform\plugins
2025-01-30T18:21:00.366+0100 [DEBUG] Found the config directory: C:\Users\Henrik\AppData\Roaming\terraform.d
2025-01-30T18:21:00.366+0100 [INFO]  CLI command args: []string{"import", "cloudflare_r2_bucket.hegerdes-mastodon", "my-id/hegerdes-mastodon", "-no-color"}
2025-01-30T18:21:00.369+0100 [DEBUG] New state was assigned lineage "12168324-ebba-4f11-26cb-c2a4c60d2b64"
2025-01-30T18:21:00.556+0100 [DEBUG] checking for provisioner in "."
2025-01-30T18:21:00.557+0100 [DEBUG] checking for provisioner in "C:\\Users\\Henrik\\Tools"
2025-01-30T18:21:00.567+0100 [DEBUG] Building and walking import graph
2025-01-30T18:21:00.567+0100 [DEBUG] Not attaching any node states: overall state is nil
2025-01-30T18:21:00.567+0100 [DEBUG] No state, no orphan outputs
2025-01-30T18:21:00.567+0100 [DEBUG] ProviderTransformer: "cloudflare_r2_bucket.hegerdes-mastodon (expand)" (*tofu.nodeExpandPlannableResource) needs provider["registry.opentofu.org/cloudflare/cloudflare"]
2025-01-30T18:21:00.567+0100 [DEBUG] created provider logger: level=debug
2025-01-30T18:21:00.568+0100 [INFO]  provider: configuring client automatic mTLS
2025-01-30T18:21:00.590+0100 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe args=[".terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe"]
2025-01-30T18:21:00.601+0100 [DEBUG] provider: plugin started: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe pid=28832
2025-01-30T18:21:00.601+0100 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe
2025-01-30T18:21:00.642+0100 [INFO]  provider.terraform-provider-cloudflare_v5.0.0.exe: configuring server automatic mTLS: timestamp="2025-01-30T18:21:00.641+0100"
2025-01-30T18:21:00.662+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: plugin address: address=127.0.0.1:10000 network=tcp timestamp="2025-01-30T18:21:00.662+0100"
2025-01-30T18:21:00.662+0100 [DEBUG] provider: using plugin: version=6
2025-01-30T18:21:00.834+0100 [DEBUG] No provider meta schema returned
2025-01-30T18:21:00.901+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-01-30T18:21:00.926+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe pid=28832
2025-01-30T18:21:00.948+0100 [DEBUG] provider: plugin exited
2025-01-30T18:21:00.951+0100 [DEBUG] ReferenceTransformer: "cloudflare_r2_bucket.hegerdes-mastodon (expand)" references: [local.account_id (expand)]
2025-01-30T18:21:00.951+0100 [DEBUG] ReferenceTransformer: "local.account_id (expand)" references: []
2025-01-30T18:21:00.951+0100 [DEBUG] ReferenceTransformer: "provider[\"registry.opentofu.org/cloudflare/cloudflare\"]" references: []
2025-01-30T18:21:00.951+0100 [DEBUG] Starting graph walk: walkImport
2025-01-30T18:21:00.953+0100 [DEBUG] created provider logger: level=debug
2025-01-30T18:21:00.953+0100 [INFO]  provider: configuring client automatic mTLS
2025-01-30T18:21:00.964+0100 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe args=[".terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe"]
2025-01-30T18:21:00.972+0100 [DEBUG] provider: plugin started: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe pid=16476
2025-01-30T18:21:00.972+0100 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe
2025-01-30T18:21:01.011+0100 [INFO]  provider.terraform-provider-cloudflare_v5.0.0.exe: configuring server automatic mTLS: timestamp="2025-01-30T18:21:01.010+0100"
2025-01-30T18:21:01.033+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: plugin address: address=127.0.0.1:10000 network=tcp timestamp="2025-01-30T18:21:01.033+0100"
2025-01-30T18:21:01.033+0100 [DEBUG] provider: using plugin: version=6
2025-01-30T18:21:01.056+0100 [DEBUG] ReferenceTransformer: "cloudflare_r2_bucket.hegerdes-mastodon (import id \"my-id/hegerdes-mastodon\")" references: []
cloudflare_r2_bucket.hegerdes-mastodon: Importing from ID "my-id/hegerdes-mastodon"...
2025-01-30T18:21:01.065+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
GET /client/v4/accounts/my-id/r2/buckets/hegerdes-mastodon HTTP/1.1
> user-agent: terraform-provider-cloudflare/5.0.0 terraform-plugin-framework/1.13.0 terraform/1.9.0
> accept: application/json
> x-stainless-runtime: terraform-plugin-framework
> x-stainless-os: Other:windows
> x-stainless-retry-count: 0
> x-stainless-runtime-version: 1.13.0
> x-auth-email: [redacted]
> x-stainless-lang: Terraform
> x-stainless-package-version: 5.0.0
> x-auth-key: [redacted]
> x-stainless-arch: x64: tf_rpc=ImportResourceState @module=cloudflare tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=61520524-0205-5fd0-0702-7c2b232bf699 tf_resource_type=cloudflare_r2_bucket @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:64 timestamp="2025-01-30T18:21:01.065+0100"
2025-01-30T18:21:01.956+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
< HTTP/2.0 200 OK
< set-cookie: __cflb=0H28vgHxwvgAQtjUGU4yFBDJQfw1pfzuEHpYTFJQBLh; SameSite=Lax; path=/; expires=Thu, 30-Jan-25 19:51:03 GMT; HttpOnly
< set-cookie: __cf_bm=6GMXXsNOtRG5qripMKOAqNiY0v_JAAJtu5AOWww0_xU-1738257662-1.0.1.1-WQXg8jeGQ13lCxTwX6WfawaBdgVJ7kr9C7pb8pkJqiS6RctADHy8tLlJxIOllcwqW52OO_KqUsfLWS_62e.g0Q; path=/; expires=Thu, 30-Jan-25 17:51:02 GMT; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: __cfruid=78cd6047babbb8efe34875ce7caaacf3ce784ae6-1738257662; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: _cfuvid=dULeg97lv9DnoO_8x5mv2i3JRj2b6lP.ILy_ZGSWR3U-1738257662078-0.0.1.1-604800000; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< vary: Accept-Encoding
< cf-auditlog-id: 0194b83a-ff07-7b61-8d36-532cf89d1c73
< server: cloudflare
< content-type: application/json; charset=UTF-8
< cf-ray: 90a322d14d8158d8-TXL
< date: Thu, 30 Jan 2025 17:21:02 GMT
< cf-cache-status: DYNAMIC
<

{"success":true,"errors":[],"messages":[],"result":{"name":"hegerdes-mastodon","creation_date":"2024-10-17T19:14:52.029Z","location":"WEUR","storage_class":"Standard"}}

: tf_provider_addr=registry.terraform.io/cloudflare/cloudflare @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:92 @module=cloudflare tf_req_id=61520524-0205-5fd0-0702-7c2b232bf699 tf_resource_type=cloudflare_r2_bucket tf_rpc=ImportResourceState timestamp="2025-01-30T18:21:01.956+0100"
cloudflare_r2_bucket.hegerdes-mastodon: Import prepared!
  Prepared cloudflare_r2_bucket for import
cloudflare_r2_bucket.hegerdes-mastodon: Refreshing state... [id=hegerdes-mastodon]
2025-01-30T18:21:01.958+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
GET /client/v4/accounts/my-id/r2/buckets/hegerdes-mastodon HTTP/1.1
> x-auth-email: [redacted]
> x-stainless-retry-count: 0
> x-stainless-lang: Terraform
> x-stainless-runtime-version: 1.13.0
> x-auth-key: [redacted]
> cf-r2-jurisdiction: 
> x-stainless-runtime: terraform-plugin-framework
> x-stainless-arch: x64
> accept: application/json
> x-stainless-os: Other:windows
> user-agent: terraform-provider-cloudflare/5.0.0 terraform-plugin-framework/1.13.0 terraform/1.9.0
> x-stainless-package-version: 5.0.0: @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:64 @module=cloudflare tf_req_id=ce9d3f82-e0e5-67d6-8789-cef1b041c5e7 tf_resource_type=cloudflare_r2_bucket tf_rpc=ReadResource tf_provider_addr=registry.terraform.io/cloudflare/cloudflare timestamp="2025-01-30T18:21:01.958+0100"
2025-01-30T18:21:02.401+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
< HTTP/2.0 500 Internal Server Error
< cf-cache-status: DYNAMIC
< cf-auditlog-id: 0194b83b-00e0-778a-b80f-00685a48c2dc
< set-cookie: __cflb=0H28vgHxwvgAQtjUGU4yFBDJQfw1pfzuETXDU47tncD; SameSite=Lax; path=/; expires=Thu, 30-Jan-25 19:51:03 GMT; HttpOnly
< set-cookie: __cf_bm=ZUSNT5BVBafTB4N_IIt9PObEt8PM.1UMhyxhFK_oEaQ-1738257662-1.0.1.1-QAU9btqRyDXJJ3xC49NY7mjfotnC_FjjXEt1VRzXoQOnTbKKNuG1zNuCkYWES2fy0plm3OLXASAFr6Rpc8oipw; path=/; expires=Thu, 30-Jan-25 17:51:02 GMT; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: __cfruid=78cd6047babbb8efe34875ce7caaacf3ce784ae6-1738257662; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: _cfuvid=u7S_zjvFCHwBuKTsn3VEJhg89PwwtfdDhmcMXvfciQ8-1738257662555-0.0.1.1-604800000; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< vary: Accept-Encoding
< server: cloudflare
< date: Thu, 30 Jan 2025 17:21:02 GMT
< content-type: text/plain;charset=UTF-8
< content-length: 13
< cf-ray: 90a322d46bc058d8-TXL
<

InternalError

: tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_resource_type=cloudflare_r2_bucket tf_req_id=ce9d3f82-e0e5-67d6-8789-cef1b041c5e7 tf_rpc=ReadResource @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:92 @module=cloudflare timestamp="2025-01-30T18:21:02.401+0100"
2025-01-30T18:21:02.829+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
GET /client/v4/accounts/my-id/r2/buckets/hegerdes-mastodon HTTP/1.1
> x-auth-key: [redacted]
> x-auth-email: [redacted]
> x-stainless-os: Other:windows
> x-stainless-lang: Terraform
> x-stainless-runtime-version: 1.13.0
> cf-r2-jurisdiction: 
> x-stainless-retry-count: 1
> user-agent: terraform-provider-cloudflare/5.0.0 terraform-plugin-framework/1.13.0 terraform/1.9.0
> x-stainless-package-version: 5.0.0
> x-stainless-runtime: terraform-plugin-framework
> accept: application/json
> x-stainless-arch: x64: @module=cloudflare tf_req_id=ce9d3f82-e0e5-67d6-8789-cef1b041c5e7 tf_rpc=ReadResource @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:64 tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_resource_type=cloudflare_r2_bucket timestamp="2025-01-30T18:21:02.829+0100"
2025-01-30T18:21:03.023+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
< HTTP/2.0 500 Internal Server Error
< set-cookie: __cflb=0H28vgHxwvgAQtjUGU4yFBDJQfw1pfzuEdDtUrwPPsj; SameSite=Lax; path=/; expires=Thu, 30-Jan-25 19:51:04 GMT; HttpOnly
< set-cookie: __cf_bm=bsVAGQjGwlDghyQJ3RSStuPSxadsoLTsihHZOH9toUk-1738257663-1.0.1.1-_QrGW3waDUBTtKS7f9dpMqqxvKYnBsVB65ESQBGVyBFWFCg3UOd3SMm8AlaOHxw3bLHSdl3MnA7sr4q2HAOhlA; path=/; expires=Thu, 30-Jan-25 17:51:03 GMT; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: __cfruid=12686e125f29af1838a1079df59c8e5bdaca546b-1738257663; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: _cfuvid=qAyzDFsxmEwNW_R4hkHAZYVtWW.CAzy43nGaebrbNnI-1738257663194-0.0.1.1-604800000; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< vary: Accept-Encoding
< content-type: text/plain;charset=UTF-8
< content-length: 13
< cf-cache-status: DYNAMIC
< server: cloudflare
< date: Thu, 30 Jan 2025 17:21:03 GMT
< cf-ray: 90a322d9ddcd58d8-TXL
< cf-auditlog-id: 0194b83b-043f-7595-a47d-349127eeef1f
<

InternalError

: tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_resource_type=cloudflare_r2_bucket @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:92 @module=cloudflare tf_req_id=ce9d3f82-e0e5-67d6-8789-cef1b041c5e7 tf_rpc=ReadResource timestamp="2025-01-30T18:21:03.023+0100"
2025-01-30T18:21:03.984+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
GET /client/v4/accounts/my-id/r2/buckets/hegerdes-mastodon HTTP/1.1
> accept: application/json
> x-stainless-lang: Terraform
> x-stainless-os: Other:windows
> x-stainless-runtime-version: 1.13.0
> user-agent: terraform-provider-cloudflare/5.0.0 terraform-plugin-framework/1.13.0 terraform/1.9.0
> x-stainless-package-version: 5.0.0
> x-auth-key: [redacted]
> cf-r2-jurisdiction: 
> x-stainless-retry-count: 2
> x-stainless-runtime: terraform-plugin-framework
> x-stainless-arch: x64
> x-auth-email: [redacted]: tf_resource_type=cloudflare_r2_bucket tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=ce9d3f82-e0e5-67d6-8789-cef1b041c5e7 tf_rpc=ReadResource @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:64 @module=cloudflare timestamp="2025-01-30T18:21:03.983+0100"
2025-01-30T18:21:04.247+0100 [DEBUG] provider.terraform-provider-cloudflare_v5.0.0.exe: 
< HTTP/2.0 500 Internal Server Error
< content-length: 13
< cf-ray: 90a322e12aaf58d8-TXL
< vary: Accept-Encoding
< cf-auditlog-id: 0194b83b-08d1-70aa-a8b8-0df6b5d1ff96
< date: Thu, 30 Jan 2025 17:21:04 GMT
< content-type: text/plain;charset=UTF-8
< server: cloudflare
< cf-cache-status: DYNAMIC
< set-cookie: __cflb=0H28vgHxwvgAQtjUGU4yFBDJQfw1pfzuEnvZVfkt19F; SameSite=Lax; path=/; expires=Thu, 30-Jan-25 19:51:05 GMT; HttpOnly
< set-cookie: __cf_bm=4VD4dsCBV2o_zPozlGUz8RZBtbFuyJqVHDN8KS7KkVw-1738257664-1.0.1.1-i_4Av64L_UiEkbNMgxRA3DxsHY3_DKCw1JRcvHez2ODSoYk45IhC2hgwm2MwTYc2bjU1S2fx4F3q10.cNYZb8g; path=/; expires=Thu, 30-Jan-25 17:51:04 GMT; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: __cfruid=bd6534fa7d5cc03d4f3f3d506b6bceda4c6b7213-1738257664; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
< set-cookie: _cfuvid=KCYKkOX4Yz.ubw3emUP53aDtlAieGWii5epcv0j5o24-1738257664342-0.0.1.1-604800000; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
<

InternalError

: @caller=github.com/cloudflare/terraform-provider-cloudflare/internal/logging/logging.go:92 @module=cloudflare tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=ce9d3f82-e0e5-67d6-8789-cef1b041c5e7 tf_resource_type=cloudflare_r2_bucket tf_rpc=ReadResource timestamp="2025-01-30T18:21:04.246+0100"
2025-01-30T18:21:04.248+0100 [ERROR] provider.terraform-provider-cloudflare_v5.0.0.exe: Response contains error diagnostic: diagnostic_detail="GET \"https://api.cloudflare.com/client/v4/accounts/my-id/r2/buckets/hegerdes-mastodon\": 500 Internal Server Error InternalError" tf_resource_type=cloudflare_r2_bucket tf_rpc=ReadResource @caller=github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_severity=ERROR diagnostic_summary="failed to make http request" tf_proto_version=6.7 tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=ce9d3f82-e0e5-67d6-8789-cef1b041c5e7 timestamp="2025-01-30T18:21:04.247+0100"
2025-01-30T18:21:04.250+0100 [ERROR] vertex "import cloudflare_r2_bucket.hegerdes-mastodon result" error: failed to make http request
2025-01-30T18:21:04.250+0100 [ERROR] vertex "cloudflare_r2_bucket.hegerdes-mastodon (import id \"my-id/hegerdes-mastodon\")" error: failed to make http request
2025-01-30T18:21:04.250+0100 [ERROR] vertex "cloudflare_r2_bucket.hegerdes-mastodon (expand)" error: failed to make http request

Error: failed to make http request

GET
"https://api.cloudflare.com/client/v4/accounts/my-id/r2/buckets/hegerdes-mastodon":
500 Internal Server Error InternalError

2025-01-30T18:21:04.257+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-01-30T18:21:04.305+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.opentofu.org/cloudflare/cloudflare/5.0.0/windows_amd64/terraform-provider-cloudflare_v5.0.0.exe pid=16476
2025-01-30T18:21:04.313+0100 [DEBUG] provider: plugin exited

  ```
  
</details>

--- a/docs/data-sources/magic_transit_site.md
+++ b/docs/data-sources/magic_transit_site.md
@@ -45,7 +45,7 @@ data "cloudflare_magic_transit_site" "example_magic_transit_site" {
 
 Optional:
 
-- `connector_identifier` (String) Identifier
+- `connectorid` (String) Identifier
 
 
 <a id="nestedatt--location"></a>

--- a/docs/data-sources/magic_transit_site_lan.md
+++ b/docs/data-sources/magic_transit_site_lan.md
@@ -92,6 +92,7 @@ Read-Only:
 - `dhcp_pool_end` (String) A valid IPv4 address.
 - `dhcp_pool_start` (String) A valid IPv4 address.
 - `dns_server` (String) A valid IPv4 address.
+- `dns_servers` (List of String)
 - `reservations` (Map of String) Mapping of MAC addresses to IP addresses
 
 

--- a/docs/data-sources/magic_transit_site_lans.md
+++ b/docs/data-sources/magic_transit_site_lans.md
@@ -102,6 +102,7 @@ Read-Only:
 - `dhcp_pool_end` (String) A valid IPv4 address.
 - `dhcp_pool_start` (String) A valid IPv4 address.
 - `dns_server` (String) A valid IPv4 address.
+- `dns_servers` (List of String)
 - `reservations` (Map of String) Mapping of MAC addresses to IP addresses
 
 

--- a/docs/data-sources/magic_transit_sites.md
+++ b/docs/data-sources/magic_transit_sites.md
@@ -14,7 +14,7 @@ description: |-
 ```terraform
 data "cloudflare_magic_transit_sites" "example_magic_transit_sites" {
   account_id = "023e105f4ecef8ad9ca31a8372d0c353"
-  connector_identifier = "023e105f4ecef8ad9ca31a8372d0c353"
+  connectorid = "023e105f4ecef8ad9ca31a8372d0c353"
 }
 ```
 
@@ -27,7 +27,7 @@ data "cloudflare_magic_transit_sites" "example_magic_transit_sites" {
 
 ### Optional
 
-- `connector_identifier` (String) Identifier
+- `connectorid` (String) Identifier
 - `max_items` (Number) Max items to fetch, default: 1000
 
 ### Read-Only

--- a/docs/data-sources/zero_trust_access_infrastructure_target.md
+++ b/docs/data-sources/zero_trust_access_infrastructure_target.md
@@ -48,13 +48,26 @@ Optional:
 - `direction` (String) The sorting direction.
 - `hostname` (String) Hostname of a target
 - `hostname_contains` (String) Partial match to the hostname of a target
+- `ip_like` (String) Filters for targets whose IP addresses look like the specified string.
+Supports `*` as a wildcard character
 - `ip_v4` (String) IPv4 address of the target
 - `ip_v6` (String) IPv6 address of the target
 - `ips` (List of String) Filters for targets that have any of the following IP addresses. Specify
 `ips` multiple times in query parameter to build list of candidates.
+- `ipv4_end` (String) Defines an IPv4 filter range's ending value (inclusive). Requires
+`ipv4_start` to be specified as well.
+- `ipv4_start` (String) Defines an IPv4 filter range's starting value (inclusive). Requires
+`ipv4_end` to be specified as well.
+- `ipv6_end` (String) Defines an IPv6 filter range's ending value (inclusive). Requires
+`ipv6_start` to be specified as well.
+- `ipv6_start` (String) Defines an IPv6 filter range's starting value (inclusive). Requires
+`ipv6_end` to be specified as well.
 - `modified_after` (String) Date and time at which the target was modified after (inclusive)
 - `modified_before` (String) Date and time at which the target was modified before (inclusive)
 - `order` (String) The field to sort by.
+- `target_ids` (List of String) Filters for targets that have any of the following UUIDs. Specify
+`target_ids` multiple times in query parameter to build list of
+candidates.
 - `virtual_network_id` (String) Private virtual network identifier of the target
 
 

--- a/docs/data-sources/zero_trust_access_infrastructure_targets.md
+++ b/docs/data-sources/zero_trust_access_infrastructure_targets.md
@@ -19,12 +19,18 @@ data "cloudflare_zero_trust_access_infrastructure_targets" "example_zero_trust_a
   direction = "asc"
   hostname = "hostname"
   hostname_contains = "hostname_contains"
+  ip_like = "ip_like"
   ip_v4 = "ip_v4"
   ip_v6 = "ip_v6"
   ips = ["string"]
+  ipv4_end = "ipv4_end"
+  ipv4_start = "ipv4_start"
+  ipv6_end = "ipv6_end"
+  ipv6_start = "ipv6_start"
   modified_after = "2019-12-27T18:11:19.117Z"
   modified_before = "2019-12-27T18:11:19.117Z"
   order = "hostname"
+  target_ids = ["182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e"]
   virtual_network_id = "182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e"
 }
 ```
@@ -43,14 +49,27 @@ data "cloudflare_zero_trust_access_infrastructure_targets" "example_zero_trust_a
 - `direction` (String) The sorting direction.
 - `hostname` (String) Hostname of a target
 - `hostname_contains` (String) Partial match to the hostname of a target
+- `ip_like` (String) Filters for targets whose IP addresses look like the specified string.
+Supports `*` as a wildcard character
 - `ip_v4` (String) IPv4 address of the target
 - `ip_v6` (String) IPv6 address of the target
 - `ips` (List of String) Filters for targets that have any of the following IP addresses. Specify
 `ips` multiple times in query parameter to build list of candidates.
+- `ipv4_end` (String) Defines an IPv4 filter range's ending value (inclusive). Requires
+`ipv4_start` to be specified as well.
+- `ipv4_start` (String) Defines an IPv4 filter range's starting value (inclusive). Requires
+`ipv4_end` to be specified as well.
+- `ipv6_end` (String) Defines an IPv6 filter range's ending value (inclusive). Requires
+`ipv6_start` to be specified as well.
+- `ipv6_start` (String) Defines an IPv6 filter range's starting value (inclusive). Requires
+`ipv6_end` to be specified as well.
 - `max_items` (Number) Max items to fetch, default: 1000
 - `modified_after` (String) Date and time at which the target was modified after (inclusive)
 - `modified_before` (String) Date and time at which the target was modified before (inclusive)
 - `order` (String) The field to sort by.
+- `target_ids` (List of String) Filters for targets that have any of the following UUIDs. Specify
+`target_ids` multiple times in query parameter to build list of
+candidates.
 - `virtual_network_id` (String) Private virtual network identifier of the target
 
 ### Read-Only

--- a/docs/data-sources/zone_cache_reserve.md
+++ b/docs/data-sources/zone_cache_reserve.md
@@ -27,8 +27,8 @@ data "cloudflare_zone_cache_reserve" "example_zone_cache_reserve" {
 ### Read-Only
 
 - `editable` (Boolean) Whether the setting is editable
+- `id` (String) ID of the zone setting.
 - `modified_on` (String) Last time this setting was modified.
 - `value` (String) The value of the feature
-- `zone_setting_id` (String) ID of the zone setting.
 
 

--- a/docs/resources/custom_hostname.md
+++ b/docs/resources/custom_hostname.md
@@ -19,6 +19,20 @@ resource "cloudflare_custom_hostname" "example_custom_hostname" {
     bundle_method = "ubiquitous"
     certificate_authority = "digicert"
     cloudflare_branding = false
+    custom_cert_bundle = [{
+      custom_certificate = <<EOT
+      -----BEGIN CERTIFICATE-----
+      MIIDdjCCAl6gAwIBAgIJAPnMg0Fs+/B0MA0GCSqGSIb3DQEBCwUAMFsx...
+      -----END CERTIFICATE-----
+
+      EOT
+      custom_key = <<EOT
+      -----BEGIN PRIVATE KEY-----
+      MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQC/SCB5...
+      -----END PRIVATE KEY-----
+
+      EOT
+    }]
     custom_certificate = "-----BEGIN CERTIFICATE-----\\nMIIFJDCCBAygAwIBAgIQD0ifmj/Yi5NP/2gdUySbfzANBgkqhkiG9w0BAQsFADBN\\nMQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMScwJQYDVQQDEx5E...SzSHfXp5lnu/3V08I72q1QNzOCgY1XeL4GKVcj4or6cT6tX6oJH7ePPmfrBfqI/O\\nOeH8gMJ+FuwtXYEPa4hBf38M5eU5xWG7\\n-----END CERTIFICATE-----\\n"
     custom_key = <<EOT
     -----BEGIN RSA PRIVATE KEY-----
@@ -99,12 +113,22 @@ Optional:
 - `bundle_method` (String) A ubiquitous bundle has the highest probability of being verified everywhere, even by clients using outdated or unusual trust stores. An optimal bundle uses the shortest chain and newest intermediates. And the force bundle verifies the chain, but does not otherwise modify it.
 - `certificate_authority` (String) The Certificate Authority that will issue the certificate
 - `cloudflare_branding` (Boolean) Whether or not to add Cloudflare Branding for the order.  This will add a subdomain of sni.cloudflaressl.com as the Common Name if set to true
+- `custom_cert_bundle` (Attributes List) Array of custom certificate and key pairs (1 or 2 pairs allowed) (see [below for nested schema](#nestedatt--ssl--custom_cert_bundle))
 - `custom_certificate` (String) If a custom uploaded certificate is used.
 - `custom_key` (String) The key for a custom uploaded certificate.
 - `method` (String) Domain control validation (DCV) method used for this hostname.
 - `settings` (Attributes) SSL specific settings. (see [below for nested schema](#nestedatt--ssl--settings))
 - `type` (String) Level of validation to be used for this hostname. Domain validation (dv) must be used.
 - `wildcard` (Boolean) Indicates whether the certificate covers a wildcard.
+
+<a id="nestedatt--ssl--custom_cert_bundle"></a>
+### Nested Schema for `ssl.custom_cert_bundle`
+
+Required:
+
+- `custom_certificate` (String) If a custom uploaded certificate is used.
+- `custom_key` (String) The key for a custom uploaded certificate.
+
 
 <a id="nestedatt--ssl--settings"></a>
 ### Nested Schema for `ssl.settings`

--- a/docs/resources/magic_transit_site_lan.md
+++ b/docs/resources/magic_transit_site_lan.md
@@ -38,6 +38,7 @@ resource "cloudflare_magic_transit_site_lan" "example_magic_transit_site_lan" {
       dhcp_pool_end = "192.0.2.1"
       dhcp_pool_start = "192.0.2.1"
       dns_server = "192.0.2.1"
+      dns_servers = ["192.0.2.1"]
       reservations = {
         "00:11:22:33:44:55" = "192.0.2.100"
         "AA:BB:CC:DD:EE:FF" = "192.168.1.101"
@@ -131,6 +132,7 @@ Optional:
 - `dhcp_pool_end` (String) A valid IPv4 address.
 - `dhcp_pool_start` (String) A valid IPv4 address.
 - `dns_server` (String) A valid IPv4 address.
+- `dns_servers` (List of String)
 - `reservations` (Map of String) Mapping of MAC addresses to IP addresses
 
 

--- a/docs/resources/r2_bucket.md
+++ b/docs/resources/r2_bucket.md
@@ -44,5 +44,5 @@ resource "cloudflare_r2_bucket" "example_r2_bucket" {
 Import is supported using the following syntax:
 
 ```shell
-$ terraform import cloudflare_r2_bucket.example '<account_id>/<bucket_name>'
+$ terraform import cloudflare_r2_bucket.example '<account_id>/<bucket_name>/<jurisdiction>'
 ```

--- a/docs/resources/zone_cache_reserve.md
+++ b/docs/resources/zone_cache_reserve.md
@@ -34,7 +34,6 @@ resource "cloudflare_zone_cache_reserve" "example_zone_cache_reserve" {
 - `editable` (Boolean) Whether the setting is editable
 - `id` (String) Identifier
 - `modified_on` (String) Last time this setting was modified.
-- `zone_setting_id` (String) ID of the zone setting.
 
 ## Import
 

--- a/examples/resources/cloudflare_r2_bucket/import.sh
+++ b/examples/resources/cloudflare_r2_bucket/import.sh
@@ -1,1 +1,1 @@
-$ terraform import cloudflare_r2_bucket.example '<account_id>/<bucket_name>'
+$ terraform import cloudflare_r2_bucket.example '<account_id>/<bucket_name>/<jurisdiction>'

--- a/internal/services/r2_bucket/resource.go
+++ b/internal/services/r2_bucket/resource.go
@@ -223,11 +223,13 @@ func (r *R2BucketResource) ImportState(ctx context.Context, req resource.ImportS
 
 	path_account_id := ""
 	path_bucket_name := ""
+	jurisdiction := "default"
 	diags := importpath.ParseImportID(
 		req.ID,
-		"<account_id>/<bucket_name>",
+		"<account_id>/<bucket_name>/<jurisdiction>",
 		&path_account_id,
 		&path_bucket_name,
+		&jurisdiction,
 	)
 	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
@@ -237,6 +239,12 @@ func (r *R2BucketResource) ImportState(ctx context.Context, req resource.ImportS
 	data.AccountID = types.StringValue(path_account_id)
 	data.Name = types.StringValue(path_bucket_name)
 
+	if jurisdiction == "default" {
+		data.Jurisdiction = types.StringValue("default")
+	} else {
+		data.Jurisdiction = types.StringValue(jurisdiction)
+	}
+
 	res := new(http.Response)
 	env := R2BucketResultEnvelope{*data}
 	_, err := r.client.R2.Buckets.Get(
@@ -245,6 +253,7 @@ func (r *R2BucketResource) ImportState(ctx context.Context, req resource.ImportS
 		r2.BucketGetParams{
 			AccountID: cloudflare.F(path_account_id),
 		},
+		option.WithHeader(jurisdictionHTTPHeaderName, data.Jurisdiction.ValueString()),
 		option.WithResponseBodyInto(&res),
 		option.WithMiddleware(logging.Middleware(ctx)),
 	)

--- a/internal/services/r2_bucket/resource_test.go
+++ b/internal/services/r2_bucket/resource_test.go
@@ -5,6 +5,7 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"strings"
 	"testing"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
@@ -15,6 +16,7 @@ import (
 	"github.com/cloudflare/terraform-provider-cloudflare/internal/acctest"
 	"github.com/cloudflare/terraform-provider-cloudflare/internal/utils"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
 )
 
 func TestMain(m *testing.M) {
@@ -120,12 +122,15 @@ func TestAccCloudflareR2Bucket_Basic(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "location", "ENAM"),
 				),
 			},
-			// {
-			// 	ResourceName:        resourceName,
-			// 	ImportStateIdPrefix: fmt.Sprintf("%s/", accountID),
-			// 	ImportState:         true,
-			// 	ImportStateVerify:   true,
-			// },
+			{
+				ResourceName: resourceName,
+				ImportStateIdFunc: func(*terraform.State) (string, error) {
+					return strings.Join([]string{accountID, rnd, "default"}, "/"), nil
+				},
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "storage_class"},
+			},
 		},
 	})
 }
@@ -171,6 +176,15 @@ func TestAccCloudflareR2Bucket_Jurisdiction(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "name", rnd),
 					resource.TestCheckResourceAttr(resourceName, "id", rnd),
 				),
+			},
+			{
+				ResourceName: resourceName,
+				ImportStateIdFunc: func(*terraform.State) (string, error) {
+					return strings.Join([]string{accountID, rnd, "eu"}, "/"), nil
+				},
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"storage_class"},
 			},
 		},
 	})

